main.cpp

```diff
@@ -91,7 +91,92 @@ good to go!
  Wait for my code review.
  */
 
+
+
+
+
 #include <iostream>
+
+struct FloatType
+{
+    float add(float lhs, float rhs )
+    {
+        return (lhs + rhs);
+    }
+    float subtract( float lhs, float rhs )
+    {
+        return (lhs - rhs);
+    }
+    float multiply( float lhs, float rhs )
+    {
+        return (lhs * rhs);
+    }
+    float divide( float lhs, float rhs )
+    {
+        if (rhs == 0)
+        {
+            std::cout << "WARNING: You are dividing by zero!" << std::endl;
+        }
+        return (lhs / rhs);
+    }
+};
+
+struct DoubleType
+{
+    double add(double lhs, double rhs )
+    {
+        return (lhs + rhs);
+    }
+    double subtract( double lhs, double rhs )
+    {
+        return (lhs - rhs);
+    }
+    double multiply( double lhs, double rhs )
+    {
+        return (lhs * rhs);
+    }
+    double divide( double lhs, double rhs )
+    {
+        if (rhs == 0)
+        {
+            std::cout << "WARNING: You are dividing by zero!" << std::endl;
+        }
+        return (lhs / rhs);
+    }
+};
+
+struct IntType
+{
+    int add(int lhs, int rhs )
+    {
+        return (lhs + rhs);
+    }
+    int subtract( int lhs, int rhs )
+    {
+        return (lhs - rhs);
+    }
+    int multiply( int lhs, int rhs )
+    {
+        return (lhs * rhs);
+    }
+    int divide( int lhs, int rhs )
+    {
+        if (rhs == 0)
+        {
+            std::cout << "WARNING: You are dividing by zero!" << std::endl;
+
+            std::cout << "Returning lhs: " << std::endl;
+            return lhs;
+        }
+
+        else
+        {
```

Comment:
don't leave random blank lines.  they make the code less readable.
- [ ] revise